client.go

```diff
@@ -1264,6 +1268,18 @@ func insertParamsFromArgsAndOptions(args JobArgs, insertOpts *InsertOpts) (*rive
 	}
 	if tags == nil {
 		tags = []string{}
+	} else {
+		for _, tag := range tags {
+			if len(tag) > 255 {
+				return nil, nil, errors.New("tags should be a maximum of 255 characters long")
+			}
+			// Restricted commas because we need those for batch inserts with
+			// the riverdatabasesql driver. We may want to restrict other
+			// special characters as well.
+			if strings.Contains(tag, ",") {
+				return nil, nil, errors.New("tags should not contain commas")
+			}
```

Comment:
maybe tags should be restricted to `/a-z_\-/i`? _possibly_ include single spaces in that but I don't think it would harm any use cases to leave that out.

Reply:
Alright, I added this variant instead:

``` go
var tagRE = regexp.MustCompile(`\A[\w][\w\-]+[\w]\z`)
```

Test matrix:

``` go
func TestTagRE(t *testing.T) {
	t.Parallel()

	require.Regexp(t, tagRE, "aaa")
	require.Regexp(t, tagRE, "_aaa")
	require.Regexp(t, tagRE, "aaa_")
	require.Regexp(t, tagRE, "777")
	require.Regexp(t, tagRE, "my-tag")
	require.Regexp(t, tagRE, "my_tag")
	require.Regexp(t, tagRE, "my_longer_tag")
	require.Regexp(t, tagRE, "My_Capitalized_Tag")
	require.Regexp(t, tagRE, "ALL_CAPS")
	require.Regexp(t, tagRE, "1_2_3")

	require.NotRegexp(t, tagRE, "a")
	require.NotRegexp(t, tagRE, "aa")
	require.NotRegexp(t, tagRE, "-aaa")
	require.NotRegexp(t, tagRE, "aaa-")
	require.NotRegexp(t, tagRE, "special@characters$banned")
	require.NotRegexp(t, tagRE, "commas,never,allowed")
}
```